modules/hooks.nix

```diff
@@ -553,10 +567,18 @@ in
           description = "Scan Nix files for dead code (unused variable bindings).";
           entry =
             let
-              toArg = string: "--" + (lib.concatMapStringsSep "-" lib.toLower (lib.filter (x: x != "") (lib.flatten (builtins.split "([[:upper:]]+[[:lower:]]+)" string))));
-              args = lib.concatMapStringsSep " " toArg (lib.filter (attr: settings.deadnix."${attr}") (lib.attrNames settings.deadnix));
+              cmdArgs =
+                mkCmdArgs (with settings.deadnix; [
+                  [ noLambdaArg "--no-lamda-arg" ]
+                  [ noLambdaPatternNames "--no-lambda-pattern-names" ]
+                  [ noUnderscore "--no-underscore" ]
+                  [ quiet "--quiet" ]
+                  [ hidden "--hidden" ]
+                  [ edit "--edit" ]
+                  [ (exclude != []) "--exclude ${concatStringsSep " -- " exclude}"]
```

Comment:
The purpose of having the `--` below was as the argument-list terminator (completely unnecessarily). Concatenating the excludes like this probably won't parse correctly given deadnix's [clap macro call](https://github.com/astro/deadnix/blob/main/src/main.rs) seems to expect you to pass the flag for each different value (not supporting a comma-separated-list like I expected), which seems weird to me (`clap` `ArgAction::get_many()`)

Reply:
> The --exclude parameter accepts multiple paths. Separate them with -- to pass [FILE_PATHS]....

In understood this as one `--exclude` parameter accepts multiple paths separated by `--`. Are you saying I'm misunderstanding that or that it is misleading?